book/src/unsafe.md

```diff
@@ -0,0 +1,32 @@
+# Unsafe Code
+
+By default, **Logos** uses unsafe code to avoid unnecessary bounds checks while
+accessing slices of the input `Source`.
+
+This unsafe code also exists in the code generated by the `Logos` derive macro,
+which generates a deterministic finite automata (DFA). Reasoning about the correctness
+of this generated code can be difficult - if the derivation of the DFA in `Logos`
+is correct, then this generated code will be correct and any mistakes in implementation
+would be caught given sufficient fuzz testing.
+
+Use of unsafe code is the default as this typically provides the fastest parser.
+
+## Disabling Unsafe Code
+
+However, for applications accepting untrusted input in a trusted context, this
+may not be a sufficient correctness justification.
+
+For those applications which cannot tolerate unsafe code, the feature `forbid-unsafe`
+may be enabled. This replaces unchecked accesses in the `Logos` crate with safe,
+checked alternatives which will panic on out-of-bounds access rather than cause
+undefined behavior. Additionally, code generated by the macro will not use the
+unsafe keyword, so generated code may be used in a crates using the 
+`#![forbid(unsafe_code)]` attribute.
+
+Disabling unsafe code will generally result in a slower parser overall, though rarely
+may end up slightly faster.
```

Comment:
Maybe mention that it depends on compiler optimizations :)

Reply:
I've reworded this section on performance to entirely remove the mention of potentially better performance.

Disabling unsafe will generally reduce throughput, so I don't want to imply that users of logos should worry about testing safe versus unsafe code to see what is faster. When the compiler does produce a surprisingly faster result, it is only for a particular micro-benchmark on a specific processor and not by a dramatic amount, so that doesn't necessarily generalize to real world usage.

Reply:
Great :-) I just think users should know that, if they really care about performances, they should benchmark the two versions and measure the trade-off in speed vs safety themselves.